app/imports/templates/components/nameStatus.js

```diff
@@ -13,6 +13,13 @@ Template['components_nameStatus'].onCreated(function() {
     try {
       registrar.getEntry(name, (err, entry) => {
         if(!err && entry) {
+          let prevInfo = TemplateVar.get(template, 'nameInfo');
```

Comment:
You have a point in that maybe both checks could be unified. I'm not liking how this piece of code is shaping up, I feel it needs some reorganization. I think the home logic could be put in a separate template called "home", and set entry.mode to "home" when !name. And as a nice side-effect, you would also see the home screen when deleting the name with backspace.